tests/migration/postcopy.go

```diff
@@ -0,0 +1,412 @@
+	"kubevirt.io/kubevirt/tests/console"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/libpod"
+	"kubevirt.io/kubevirt/tests/libstorage"
+)
+
+var _ = SIGMigrationDescribe("VM Post Copy Live Migration", func() {
+	var (
+		virtClient kubecli.KubevirtClient
+		err        error
+	)
+
+	BeforeEach(func() {
+		checks.SkipIfMigrationIsNotPossible()
+		virtClient = kubevirt.Client()
+	})
+
+	waitUntilMigrationMode := func(vmi *v1.VirtualMachineInstance, expectedMode v1.MigrationMode, timeout int) *v1.VirtualMachineInstance {
+		By("Waiting until migration status")
+		EventuallyWithOffset(2, func() v1.MigrationMode {
+			By("Retrieving the VMI post migration")
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			if vmi.Status.MigrationState != nil {
+				return vmi.Status.MigrationState.Mode
+			}
+			return v1.MigrationPreCopy
+		}, timeout, 1*time.Second).Should(Equal(expectedMode), fmt.Sprintf("migration should be in %s after %d s", expectedMode, timeout))
+		return vmi
+	}
+
+	Describe("Starting a VirtualMachineInstance ", func() {
+
+		Context("migration postcopy", func() {
+
+			var migrationPolicy *migrationsv1.MigrationPolicy
+
+			BeforeEach(func() {
+				By("Allowing post-copy and limit migration bandwidth")
+				policyName := fmt.Sprintf("testpolicy-%s", rand.String(5))
+				migrationPolicy = kubecli.NewMinimalMigrationPolicy(policyName)
+				migrationPolicy.Spec.AllowPostCopy = kvpointer.P(true)
+				migrationPolicy.Spec.CompletionTimeoutPerGiB = kvpointer.P(int64(1))
+				migrationPolicy.Spec.BandwidthPerMigration = kvpointer.P(resource.MustParse("5Mi"))
+			})
+
+			Context("with datavolume", func() {
+				var dv *cdiv1.DataVolume
+
+				BeforeEach(func() {
+					sc, foundSC := libstorage.GetRWXFileSystemStorageClass()
+					if !foundSC {
+						Skip("Skip test when Filesystem storage is not present")
+					}
+
+					dv = libdv.NewDataVolume(
+						libdv.WithRegistryURLSourceAndPullMethod(cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskFedoraTestTooling), cdiv1.RegistryPullNode),
+						libdv.WithPVC(
+							libdv.PVCWithStorageClass(sc),
+							libdv.PVCWithVolumeSize(cd.FedoraVolumeSize),
+							libdv.PVCWithReadWriteManyAccessMode(),
+						),
+					)
+
+					dv, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(testsuite.NamespacePrivileged).Create(context.Background(), dv, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				AfterEach(func() {
+					libstorage.DeleteDataVolume(&dv)
+				})
+
+				It("[test_id:5004] should be migrated successfully, using guest agent on VM with postcopy", func() {
+					VMIMigrationWithGuestaAgent(virtClient, dv.Name, "1Gi", migrationPolicy)
+				})
+
+			})
+
+			Context("[Serial] should migrate using for postcopy", func() {
```

Comment:
This `[Serial]` is misleading since actually one of the `Entry` of the `DescribeTable` inside of it will run in parallel due to the missing `Serial` decorator.

Reply:
Thanks. Yes, the top level [Serial] was a mistake, it wasn't actually `Serial`